rules-reviewed/openshift/mail.windup.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0"?>
+<ruleset id="mail"
+    xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This is a ruleset for detecting Mail API usage when migrating an application to a cloud environment.
+        </description>
+        <dependencies>
+          <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final"/>
+          <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="java"/>
+        <sourceTechnology id="java-ee"/>
+        <targetTechnology id="cloud-readiness"/>
+        <tag>mail</tag>
+    </metadata>
+    <rules>
+        <rule id="mail-00000">
+            <when>
+                    <javaclass references="javax.mail.{*}">
+                        <location>CONSTRUCTOR_CALL</location>
+                        <location>IMPORT</location>
+                        <location>INHERITANCE</location>
+                        <location>VARIABLE_DECLARATION</location>
+                    </javaclass>
+            </when>
+            <perform>
+              <hint title="JavaMail: direct access of an SMTP server" effort="3" category-id="cloud-optional">
+                  <message>
+                    Better avoid to connect to an SMTP server `host:port` to send an mail represents a direct connection with poor results in terms of load balancing and availability.  
+                    Evaluate to switch to a dedicated service for mail delivery called through a messaging system or ReST API.
```

Comment:
Yep, the "e" I think it's just for pronunciation otherwise "rst" would have looked only as 3 consecutive letters of the alphabet ;-)
I looked at [JAX-RS JSR-311](https://jcp.org/en/jsr/detail?id=311) and it's always written all in capital letters so "REST" is good.